server/services/zigbee2mqtt/lib/setValue.js

```diff
@@ -0,0 +1,53 @@
+const { BadParameters } = require('../../../utils/coreErrors');
+const logger = require('../../../utils/logger');
+
+/**
+ * @description Set the new device value from Gladys to MQTT.
+ * @param {Object} device - Updated Gladys device.
+ * @param {Object} deviceFeature - Updated Gladys device feature.
+ * @param {string|number} value - The new device feature value.
+ * @example
+ * setValue(device, deviceFeature, 0);
+ */
+function setValue(device, deviceFeature, value) {
+  const externalId = device.external_id;
+
+  if (!externalId.startsWith('zigbee2mqtt:')) {
+    throw new BadParameters(
+      `Zigbee2mqtt device external_id is invalid : "${externalId}" should starts with "zigbee2mqtt:"`,
+    );
+  }
+  // Remove first 'zigbee2mqtt:' substring
+  const topic = externalId.substring(12);
+  if (topic.length === 0) {
+    throw new BadParameters(`Zigbee2mqtt device external_id is invalid : "${externalId}" have no MQTT topic`);
+  }
+
+  // Convert Gladys value to Zigbee value
+  let zigbeeValue;
+  switch (deviceFeature.type) {
+    case 'binary':
+      zigbeeValue = value ? `{"state": "ON"}` : `{"state": "OFF"}`;
+      break;
+    case 'brightness':
+      zigbeeValue = `{"brightness": ${value}}`;
```

Comment:
For information, you can `JSON.stringify` to transform objects to string.
It can be useful for "big" objects. Here it's fine.